.github/workflows/build-publish-staging.yaml

```diff
@@ -88,4 +91,44 @@ jobs:
             push: ${{ github.event_name != 'pull_request' }}
             tags: ${{ steps.meta.outputs.tags }}
             labels: ${{ steps.meta.outputs.labels }}
-            annotations: ${{ steps.meta.outputs.annotations }}
\ No newline at end of file
+            annotations: ${{ steps.meta.outputs.annotations }}
+
+    update-helm-charts:
+      needs: [build-preview]
+      environment: preview
+      runs-on: ubuntu-latest
+      permissions:
+        contents: write
+        pull-requests: write
+
+      steps:
+        - name: Create GitHub App Token
+          id: app-token
+          uses: actions/create-github-app-token@v1
+          with:
+            app-id: ${{ secrets.APP_ID }}
+            private-key: ${{ secrets.APP_KEY }}
+        - name: Checkout Charts Repo
+          uses: actions/checkout@v4
+          with: 
+            repository: "${{ vars.DOCS_NEXT_CHARTS_ORG }}/${{ vars.DOCS_NEXT_CHARTS_REPO }}"
+            token: ${{ steps.app-token.outputs.token }}
+        - name: Commit Changes
+          env:
+            image: ${{ vars.REGISTRY }}/${{ vars.REGISTRY_ORG }}/${{ vars.IMG_NAME }}
+            tag: ${{ needs.build-preview.outputs.image_version }}
+          run: |
+            git config --global user.name 'github-actions[bot]'
+            git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+            sed -i 's/^tag: .*/tag: ${{ env.tag }}/' ./charts/docusaurus/values-stg.yaml
+            sed -i 's/^image: .*/image: ${{ env.image }}/' ./charts/docusaurus/values-stg.yaml
+            git commit -am "Automatic commit from GitHub Actions triggered by action #${{github.run_number}}"
```

Comment:
Isn't a push to a branch required before creating a pull request in the next step?

Reply:
this github action is "supposed" to work like that. I have the same concerns as you; this PR is a smoke test to check this action if it working as intended